src/typegoose.ts

```diff
@@ -69,6 +59,23 @@ export class Typegoose {
     return models[name] as ModelType<this> & T;
   }
 
+  public buildPublicSchema<T>(t: T, { existingMongoose, schemaOptions }: GetModelForClassOptions = {}) {
+    const name = this.constructor.name;
+
+    // get schema of current model
+    let sch = this.buildSchema<T>(t, name, schemaOptions);
```

Comment:
(same here)
i would recommend renaming `sch` to `schema` (i know that it was already named so)